Refactor: External TeX Live docker container

b0ink opened the pull request:
**Depends on https://github.com/thoth-tech/doubtfire-api/pull/38**
Feasibility document: https://github.com/thoth-tech/documentation/pull/534

# Description
This update takes the TeX Live installation out of the current `dev.Dockerfile` and creates a new container using the [official TeX Live Docker image](https://hub.docker.com/r/texlive/texlive). This will improve the process of building and deploying the main image as it no longer requires downloading and installing TeX Live.

Similar to the main container, the texlive container is kept alive using
```yml
command: /bin/sh -c "while sleep 5000; do :; done"
```
in the docker-compose setup, allowing doubtfire-api to `docker exec` into this container without needing to spin a new container up.

The [latex_build.sh](https://github.com/b0ink/doubtfire-api/blob/a8dead91f7866f090d3fd314f01ca6e1c58cf728/lib/shell/latex_build.sh) script is copied from doubtfire-api into the TeX Live container, and executed in the `docker exec` command, which will then run lualatex.

Two additional environment variables are defined in `devcontainer.env`
```env
LATEX_CONTAINER_NAME=formatif-texlive-container
LATEX_BUILD_PATH=/texlive/shell/latex_build.sh
```

These env variables are used in [latex_run.sh](https://github.com/b0ink/doubtfire-api/blob/ce988b26de5b18a919ff980f33cf08aff584bbeb/lib/shell/latex_run.sh), and validated in the [latex_helper.rb](https://github.com/b0ink/doubtfire-api/blob/ce988b26de5b18a919ff980f33cf08aff584bbeb/app/helpers/latex_helper.rb) module, which requires the name of the TeX Live container to `docker exec` into, and the path to where the `latex_build.sh` is stored on the TeX Live container, as this is defined in [texlive.Dockerfile](https://github.com/thoth-tech/doubtfire-deploy/pull/13/files#diff-a5e9848729ec0b2990016712595f0f407b5e2b9cafef656940396cc2e02d79e6R11-R12)

b0ink committed:
refactor: initial setup for texlive container

b0ink committed:
fix: create texlive dockerfile with dependencies

b0ink committed:
fix: mount api test files to fix student-work symlinks

b0ink committed:
fix: remove workspace mounts, use docker-compose for shared volumes

- allows shared volumes to work when running containers through vscode dev container

b0ink committed:
refactor: format

b0ink committed:
fix: ensure dev container initialises first

b0ink committed:
fix: copy script instead of using a mount and fix permissions

b0ink committed:
chore: create env variables for latex container

b0ink committed:
chore: update path to script

b0ink committed:
chore: rename latex build script

b0ink committed:
chore: remove unnecessary volume mount

b0ink committed:
Merge remote-tracking branch 'origin/development' into refactor/latex-docker-downsize-deploy

b0ink committed:
refactor: use original rails-latex tmp folder

JackSCarroll commented:
This looks good and works as intended. 
The latex container is set up correctly and added to the compose with appropriate shared volumes. 
A possible modification to docker-compose.yml is to change `while sleep 5000; do :; done` to `sleep infinity`.

b0ink committed:
fix: ensure devcontainer user has docker permissions

macite committed:
chore: update submodules

macite committed:
chore: switch settings to port 4200

macite committed:
Merge branch 'development' into 8.0.x

macite committed:
chore: keep wip files

macite committed:
Merge branch '8.0.x' of https://github.com/doubtfire-lms/doubtfire-deploy into 8.0.x

macite committed:
fix: update images for dev container and compose

- Ensure devcontainer has svg utils
- Update compose to use the latest development image from each project

macite committed:
Merge branch 'development' into 8.0.x

macite committed:
chore: add dev container build action

macite committed:
fix: change env in deploy action

macite committed:
fix: add missing docker file reference

macite committed:
fix: dev container image remove package file

macite committed:
fix: ensure action recursive checkout

macite committed:
feat: add multiplatform build

macite committed:
fix: remove symlink to doubtfire in devcontainer

macite committed:
fix: rework build yml

macite committed:
fix: correct env setting in dev dockerfile

macite committed:
Merge branch 'new/compose-container' into refactor/latex-docker-downsize-deploy

b0ink commented:
Merged in 10.0.x upstream
